ratis-server/src/main/java/org/apache/ratis/statemachine/impl/SimpleStateMachineStorage.java

```diff
@@ -228,6 +228,17 @@ public SingleFileSnapshotInfo getLatestSnapshot() {
     }
   }
 
+  public void loadLatestSnapshot() {
```

Comment:
Let's make it return `SingleFileSnapshotInfo` and then reuse it in `getLatestSnapshot()`.
```java
diff --git a/ratis-server/src/main/java/org/apache/ratis/statemachine/impl/SimpleStateMachineStorage.java b/ratis-server/src/main/java/org/apache/ratis/statemachine/impl/SimpleStateMachineStorage.java
index 88cc57dab5..7e8afbaa85 100644
--- a/ratis-server/src/main/java/org/apache/ratis/statemachine/impl/SimpleStateMachineStorage.java
+++ b/ratis-server/src/main/java/org/apache/ratis/statemachine/impl/SimpleStateMachineStorage.java
@@ -217,6 +217,10 @@ public class SimpleStateMachineStorage implements StateMachineStorage {
     if (s != null) {
       return s;
     }
+    return loadLatestSnapshot();
+  }
+
+  public SingleFileSnapshotInfo loadLatestSnapshot() {
     final File dir = stateMachineDir;
     if (dir == null) {
       return null;
```

Reply:
Done